source/extended-json.rst

```diff
@@ -0,0 +1,618 @@
+* Types in *italics* are defined elsewhere in the table or in the `JSON
+  specification`_.
+* Types in <angle brackets> detail the contents of a value.
+
++--------------------+----------------------------------------------------------+
+|**BSON 1.1 Type or  |**Extended JSON Format**                                  |
+|Convention**        |                                                          |
++--------------------+----------------------------------------------------------+
+|ObjectId            |{"$oid": <ObjectId bytes as 23-character, big-endian *hex |
+|                    |string* [#]_>                                             |
++--------------------+----------------------------------------------------------+
+|Symbol              |{"$symbol": *string*}                                     |
++--------------------+----------------------------------------------------------+
+|String              |*string*                                                  |
++--------------------+----------------------------------------------------------+
+|Int32               |{"$numberInt": <32-bit signed integer as a *string*>      |
++--------------------+----------------------------------------------------------+
+|Int64               |{"$numberLong": <64-bit signed integer as a *string*>     |
++--------------------+----------------------------------------------------------+
+|Double              |{"$numberDouble": <64-bit binary floating point as a      |
+|                    |*string*>                                                 |
++--------------------+----------------------------------------------------------+
+|Decimal128          |{"$numberDecimal": <decimal as a *string*>} [#]_          |
++--------------------+----------------------------------------------------------+
+|Binary              |{"$binary": <base64-encoded (with padding as ``=``)       |
+|                    |payload as a *string*, "$type": <BSON binary type as a    |
+|                    |one- or two-character *hex string*}                       |
++--------------------+----------------------------------------------------------+
+|Code                |{"$code": *string*}                                       |
++--------------------+----------------------------------------------------------+
+|CodeWScope          |{"$code": *string*, "$scope": *Document*}                 |
++--------------------+----------------------------------------------------------+
+|Document            |*object* (with Extended JSON extensions)                  |
++--------------------+----------------------------------------------------------+
+|Timestamp           |{"$timestamp": <64-bit unsigned integer as a *string*>}   |
++--------------------+----------------------------------------------------------+
+|Regex               |{"$regex": *string*, "$options": <BSON regex options as a |
+|                    |*string* or "" [#]_}                                      |
```

Comment:
Missing symbol fixed.

> Should we clarify that the $regex string does not include delimiters?

What do you mean? The `/` characters that denote a Javascript regex? I don't think it's necessary to clarify that, since `$regex` maps to a string, not a Javascript regex object.

> Should we clarify that characters in $options should be ordered alphabetically when generated extended JSON?

Already required. See note at the bottom of the table.

> Should we permit the $options field be optional? If not during generation then perhaps during parsing?

This spec doesn't prevent parsers from parsing regex JSON without `$options` present. However, the spec does require `$options` for consistency.